jvm-packages/xgboost4j-spark-gpu/src/main/scala/org/apache/spark/GpuParallelismTracker.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ Copyright (c) 2014 by Contributors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package org.apache.spark
+
+import org.apache.commons.logging.LogFactory
+
+/**
+ * A tracker that ensures enough number of executors are alive.
+ * Throws an exception when the number of alive executors is less than nWorkers.
+ *
+ * @param sc The SparkContext object
+ * @param timeout The maximum time to wait for enough number of workers.
+ * @param numWorkers nWorkers used in an XGBoost Job
+ */
+class GpuParallelismTracker(override val sc: SparkContext, timeout: Long, numWorkers: Int)
```

Comment:
What's the difference between CPU and GPU tracker?

Reply:
CPU and GPU will wait for different resources in Spark, like, cpu is waiting for cpu cores while gpu will wait for the workers.